src/main/java/com/linecorp/armeria/client/ClientInvocationHandler.java

```diff
@@ -117,22 +117,19 @@ private Object invokeClientMethod(Method method, Object[] args) throws Throwable
                 return resultFuture.sync().getNow();
             }
         } catch (Throwable cause) {
-            final Throwable finalCause;
-            if (cause instanceof ClosedChannelException) {
-                finalCause = ClosedSessionException.INSTANCE;
-            } else if (cause instanceof Error ||
-                       cause instanceof RuntimeException ||
-                       Stream.of(method.getExceptionTypes()).anyMatch(v -> v.isInstance(cause))) {
-                finalCause = cause;
-            } else {
-                finalCause = new UndeclaredThrowableException(cause);
-            }
-
-            throw finalCause;
+            throw simplifyCause(cause);
         }
     }
 
     private EventLoop eventLoop() {
         return ServiceInvocationContext.mapCurrent(ServiceInvocationContext::eventLoop, eventLoopGroup::next);
     }
+
+    private static Throwable simplifyCause(Throwable cause) {
```

Comment:
It was originally slightly more complicated than this. Yeah, let's remove the cruft.